main.go

```diff
@@ -240,7 +239,9 @@ func main() {
 	managerOpts.KubeConfig.UserAgent = remote.DefaultClusterAPIUserAgent(controllerName)
 
 	if watchNamespace != "" {
-		managerOpts.Cache.Namespaces = []string{watchNamespace}
+		managerOpts.Cache.DefaultNamespaces = map[string]cache.Config{
+			watchNamespace: {},
+		}
 	}
 
 	if profilerAddress != "" && enableContentionProfiling {
```

Comment:
```suggestion
	if enableContentionProfiling {
```
Just to align to what we changed in core CAPI.